### ISSUE TITLE
Expose Language Features

### ISSUE THREAD

**Issue by StefanRetief** [microsoft/monaco-editor#2043]
I've done this personally for my own project, but I think that it would be useful to expose language features from the different monaco-* packages. For example, in my project in particular, I've implemented a similar "problems" list to vscode. I cloned the monaco-typescript project and exposed the features, which has helped for instances where I wanted to use the ```DiagnosticsAdaptor``` or the ```CodeActionAdaptor``` instead of implementing essentially the same logic. Currently, I don't believe there is a method to import/utilize these methods in the current monaco-editor.

I can submit a pull request for monaco-typescript if it is found useful for others.

**Comment by rcjsuen** [comment]
@StefanRetief I think the demand is there given #1885 though I'm not positive if what you are proposing is the same as what's described in that other issue.

**Comment by StefanRetief** [comment]
@rcjsuen I believe this might be slightly different? From what I can tell (and I could be completely off here), they are describing being able to communicate directly with the worker here: https://github.com/microsoft/monaco-html/blob/master/src/htmlWorker.ts. I'm proposing something a little different, as exposing the methods/classes here: https://github.com/microsoft/monaco-html/blob/master/src/languageFeatures.ts

**Comment by rcjsuen** [comment]
@StefanRetief Yes, I see what you mean. The two pieces are related but serve two different purposes.
